### ISSUE TITLE
Diagram visualizing how Redux-ORM works

### ISSUE THREAD

**Issue by haveyaseen** [tommikaikkonen/redux-orm#203]
Newcomers want to quickly understand how Redux-ORM does its job so that

- they can more easily make the decision to start using the library and
- so they know where in the codebase which issue may need to be solved.

I'm thinking of a diagram similar to this

![image](https://user-images.githubusercontent.com/6516575/40828256-6ca98742-6580-11e8-892c-bc686def030d.png)

taken directly from [the Flux docs](https://facebook.github.io/flux/docs/in-depth-overview.html#content).

**Comment by henrymoulton** [comment]
Might pick this up since I'm using Redux ORM for the first time and introducing it to other devs I'm working with. 

I've already created a diagram similar to the above that shows where Redux ORM integrates with Redux in the 'flow'. 

I noticed that the Redux ORM Primer https://github.com/tommikaikkonen/redux-orm-primer hadn't been updated, it seems that @markerikson has been leading the way with Redux ORM documentation through his Practical Redux series and his Project Mini Mek: https://github.com/markerikson/project-minimek-educative

I felt that perhaps we could take the Redux ORM Primer, remove the out of date content and code and replace it with 

1. A diagram like you suggested
2. A project similar to Mark's but more focused on Redux ORM.

**Comment by haveyaseen** [comment]
Yeah, the Redux-ORM Primer could use an overhaul. Unfortunately Tommi isn't really invested in Redux-ORM anymore, so we will probably have to do something on our own.

That being said, I would be more than happy to include graphics in the wiki, and potentially in the README as well! Everything that helps users understand the library is welcome :slightly_smiling_face:

**Comment by henrymoulton** [comment]
![Initial Diagram](https://user-images.githubusercontent.com/30093692/42296806-b8a10efa-7ff1-11e8-9e31-86f477b6e59c.png)

Draft of the diagram I created, still lots of Redux ORM specific stuff to include but I thought beginners might want to have some clarity on Redux ORM NOT handling data fetching.

Open to ideas on what the Diagram should include.

**Comment by plandem** [comment]
as a newcomer i would probably be still puzzled, but the same time - i have no any idea how simplify idea behind :( Great work!

**Comment by haveyaseen** [comment]
I like the fact that you explained the one-way data flow so explicitly. What is missing from your diagram is that you just have Redux-ORM's reducer be called once and it automatically calls all registered models' reducers for you.

That is a good first step. Another diagram would be useful to see which steps are required to set up a session with all the necessary method calls and model subclasses.

**Comment by henrymoulton** [comment]
> as a newcomer i would probably be still puzzled, but the same time - i have no any idea how simplify idea behind :( Great work!

> That is a good first step. Another diagram would be useful to see which steps are required to set up a session with all the necessary method calls and model subclasses.

I think both of these comments are connected - the diagram is yet to show off setting up Redux ORM models which will probably be helpful.

Also, when writing about setting up Models and relations I imagine it would be good to highlight some resources for more 'pure' Front-End Developers to read about relations, RDMS, querying. 

Once this is done I think guiding the user to the idea that actually having a normalised 'little' database in your Redux store could be a good idea.

**Comment by markerikson** [comment]
I've got some relevant links and sections on that topic in the [Structuring Reducers](https://redux.js.org/recipes/structuring-reducers) section of the Redux docs.

**Comment by haveyaseen** [comment]
Some key concepts to convey for beginners would be:

- Each model needs to be a `class extends Model {}`.
- `createReducer` ensures that each `Model`'s `static reducer(action, model, session)` is called upon `dispatch`.
- `createSelector` ensures that passed functions will get re-executed if and only if a `Model.reducer` changes at least one relevant `Model`'s table (or row, in the case of instance-based queries).
